sandbox/lua/encoders/schema_influx_write.lua

```diff
@@ -0,0 +1,319 @@
+local function sub_func(key)
+    if base_fields_map[key] then
+        return read_message(key)
+    else
+        local val = read_message("Fields["..key.."]")
+        if val then
+            return val
+        end
+        return "%{"..key.."}"
+    end
+end
+
+-- Remove blacklisted fields from the set of base fields that we use, and
+-- create a table of dynamic fields to skip.
+local used_base_fields = {}
+local skip_fields_str = read_config("skip_fields")
+local skip_fields = {}
+if skip_fields_str then
+    for field in string.gmatch(skip_fields_str, "[%S]+") do
+        skip_fields[field] = true
+    end
+
+    for _, base_field in ipairs(base_fields_list) do
+        if skip_fields["**all_base**"] then
+            skip_fields[base_field] = true
+        elseif not skip_fields[base_field] then
+            used_base_fields[#used_base_fields+1] = base_field
+        else
+            skip_fields[base_field] = nil
+        end
+    end
+else
+    used_base_fields = base_fields_list
+end
+
+-- Create and populate a table of fields to be used as tags
+local tag_fields_str = read_config("tag_fields") or "**all_base**"
+local used_tag_fields = {}
+if tag_fields_str then
+    for field in string.gmatch(tag_fields_str, "[%S]+") do
+        used_tag_fields[field] = true
+    end
+end
+
+function process_message()
+    local columns = {}
+    local values = {}
+
+    -- Determine the precision of the timestamp and do the math
+    local timestamp_divisor = 1
+    if timestamp_precision == "u" then
+        timestamp_divisor = 1e3
+    elseif timestamp_precision == "ms" then
+        timestamp_divisor = 1e6
+    elseif timestamp_precision == "s" then
+        timestamp_divisor = 1e9
+    elseif timestamp_precision == "m" then
+        timestamp_divisor = 1e9 * 60
+    elseif timestamp_precision == "h" then
+        timestamp_divisor = 1e9 * 60 * 60
+    end
+    local message_timestamp = math.floor(read_message("Timestamp") / timestamp_divisor)
+
+    -- Create a counter for the table indexes for columns and values
+    -- then populate those tables with the used base fields previously
+    -- determined
+    local place = 1
+    for _, field in ipairs(used_base_fields) do
+        columns[place] = field
+        values[place] = read_message(field)
+        place = place + 1
+    end
+
+    -- If the %{field} substitutions are defined in the name_prefix,
+    -- replace them with the actual values from the message here
+    if use_subs then
+        name_prefix = string.gsub(name_prefix_orig, "%%{([%w%p]-)}", sub_func)
+    else
+        name_prefix = name_prefix_orig
+    end
+
+    -- Initialize and populate the table of tags to include in the
+    -- InfluxDB write API message
+    -- Convert value to a string as this is required by the API
+    local message_tags = {}
+    for _, field in ipairs(base_fields_list) do
+        if (used_tag_fields["**all_base**"] or used_tag_fields["**all_base**"])
```

Comment:
Looking at this further, it occurs to me that you know whether the `**all_base**` and `**all**` keys exist at config loading time. We can convert these key lookups to boolean values and save a few cycles, rather than having to perform the key lookup multiple times inside this loop for every message.